src/rantanplan/core.py

```diff
@@ -681,21 +743,27 @@ def _get_scansion(text, rhyme_analysis=False, rhythm_format="pattern",
         if rhythmical_lengths is not None:
             structure_length = rhythmical_lengths
         else:
+            # Handle repetitive stanzas
             line_structure = line.get("structure", None)
-            structure_length = STRUCTURES_LENGTH.get(line_structure, None)
-        if structure_length is not None:
+            structure_length, repetitive_structure = STRUCTURES_LENGTH.get(
+                line_structure, [[], False])
+            if structure_length and repetitive_structure:
+                repetitions = int(len(lines) / len(structure_length))
+                structure_length = structure_length * repetitions
+        if structure_length:
             if line["rhythm"]["length"] < structure_length[idx]:
                 candidates = generate_phonological_groups(raw_tokens[idx])
                 for candidate in candidates:
                     rhythm = get_rhythmical_pattern(
-                        candidate, rhythm_format)
+                        candidate, rhythm_format,
+                        rhyme_analysis=rhyme_analysis)
                     if rhythm["length"] == structure_length[idx]:
                         line.update({
                             "phonological_groups": candidate,
                             "rhythm": rhythm,
                         })
                         break
-    return lines
+    return remove_exact_length_matches(lines)
```

Comment:
I know we've discussed this, but I was wondering about the time impact of this function, which iterates over all lines, when compared to the time needed to check if `length_range` is present where appropriate.

Reply:
Yes you are right but remember our product manager wants that removed so I think it's better to leave it that way and try to convince him, and if he agrees I can make the appropiate changes. Impact on tests should be minimal if we change it to what you say.

Reply:
![alt text](https://media3.giphy.com/media/ZC0ATzzJnKqn2SNDHR/giphy.gif?cid=ecf05e47cca5cae8f6ed7430364213dc20e3abdffe09d105&rid=giphy.gif)